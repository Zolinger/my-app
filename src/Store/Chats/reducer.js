import { ADD_CHAT_ACTION } from "./constants";

const initialState = {
  chatList: [],
};

export const chatsReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_CHAT_ACTION: {
      return {
        ...state,
        chatList: [
          ...state.chatList,
          { id: "uniqueId", name: action.payload.name },
        ],
      };
    }
    default:
      return state;
  }
};
