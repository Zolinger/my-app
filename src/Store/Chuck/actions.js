import { CHUCK_URL } from "../../API/index";
import {
  GET_CHUCK_REQUEST,
  GET_CHUCK_SUCCESS,
  GET_CHUCK_ERROR,
} from "./constants";

export const getChuckRequestAction = () => ({
  type: GET_CHUCK_REQUEST,
});

export const getChuckSuccessAction = (chuckUrl) => ({
  type: GET_CHUCK_SUCCESS,
  payload: chuckUrl,
});

export const getChuckErrorAction = () => ({
  type: GET_CHUCK_ERROR,
});

export const getChuckActionWithThunk = () => async (dispatch) => {
  dispatch(getChuckRequestAction());

  try {
    const response = await fetch(CHUCK_URL);

    if (!response.ok) {
      throw new Error(`error getting data with status`);
    }

    const result = await response.json();

    dispatch(getChuckSuccessAction(result));
  } catch (error) {
    dispatch(getChuckErrorAction());
  }
};
