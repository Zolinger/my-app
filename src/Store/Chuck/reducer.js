import {
  GET_CHUCK_REQUEST,
  GET_CHUCK_SUCCESS,
  GET_CHUCK_ERROR,
} from "./constants";

const initialState = {
  chuckUrl: undefined,
  loading: true,
  error: false,
};

export const chuckReducer = (state = initialState, action) => {
  switch (action.type) {
    case GET_CHUCK_REQUEST: {
      return {
        ...state,
        loading: true,
        error: false,
      };
    }
    case GET_CHUCK_SUCCESS: {
      return {
        ...state,
        loading: false,
        chuckUrl: action.payload,
      };
    }
    case GET_CHUCK_ERROR: {
      return {
        ...state,
        loading: false,
        error: true,
      };
    }
    default:
      return state;
  }
};
