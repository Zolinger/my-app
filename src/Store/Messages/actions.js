import { ADD_MESSAGE_ACTION } from "./constants";

export const addMessageAction = (payload) => ({
  type: ADD_MESSAGE_ACTION,
  payload,
});
