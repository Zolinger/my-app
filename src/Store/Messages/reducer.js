import { ADD_MESSAGE_ACTION } from "./constants";
import { v4 as uuidv4 } from "uuid";

const initialState = {
  messageList: {},
};

export const messagesReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_MESSAGE_ACTION: {
      const { chatId, ...rest } = action.payload;

      const chatMessages = state.messageList[chatId] ?? [];
      return {
        ...state,
        messageList: {
          ...state.messageList,
          [chatId]: [...chatMessages, { id: uuidv4(), ...rest }],
        },
      };
    }
    default:
      return state;
  }
};
