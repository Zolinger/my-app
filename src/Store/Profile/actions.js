import { TOGGLE_USER_NAME_ACTION, CHANGE_USER_NAME_ACTION } from "./constants";

export const toggleUserNameAction = () => ({
  type: TOGGLE_USER_NAME_ACTION,
});

export const changeUserNameAction = (payload) => ({
  type: CHANGE_USER_NAME_ACTION,
  payload,
});
