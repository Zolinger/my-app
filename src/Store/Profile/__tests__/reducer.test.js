import { toggleUserNameAction } from "../actions";
import { profileReducer } from "../reducer";

describe("test profile reducer", () => {
  it("test toggleUserNameAction", () => {
    const received = profileReducer(undefined, toggleUserNameAction());

    expect(received).toMatchSnapshot();
  });
});
