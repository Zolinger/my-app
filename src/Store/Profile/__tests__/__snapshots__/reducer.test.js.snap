// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test profile reducer test toggleUserNameAction 1`] = `
Object {
  "name": "Alex",
  "showName": false,
}
`;
